pkg/discovery/worker/garbage_collector.go

```diff
@@ -231,3 +235,70 @@ func (g *GarbageCollector) unpauseRollouts() error {
 
 	return nil
 }
+
+func (g *GarbageCollector) cleanupQuotas() error {
+	quotaList, err := g.client.CoreV1().ResourceQuotas("").List(context.TODO(), gcListOpts)
+	if err != nil {
+		glog.Errorf("Error garbage cleaning quotas: %v", err)
+		return err
+	}
+	if quotaList == nil {
+		// ignore
+		return nil
+	}
+
+	for _, quota := range quotaList.Items {
+		err := revertQuota(g.client, &quota)
+		if err != nil {
+			// we force kubeturbo to restart
+			return err
+		}
+	}
+
+	return nil
+}
+
+func revertQuota(client *kubernetes.Clientset, quota *api.ResourceQuota) error {
+	var revertedQuota *api.ResourceQuota
+	var err error
+	if quota.Annotations != nil {
+		origSpec, exists := quota.Annotations[executor.QuotaAnnotationKey]
+		if exists {
+			revertedQuota, err = executor.DecodeQuota([]byte(origSpec))
+			if err != nil {
+				// This is very unlikely but not an error which we can recover from, for example in the next run.
+				// We need to ignore this. The annotation if there still is any would then carry the opportunity
+				// for an user to use this as a reference in case a manual correction is required ever.
+				glog.Warningf("Error reverting quota while garbage collecting resources: %s/%s: %v", quota.Namespace, quota.Name, err)
```

Comment:
I don't quite understand this. If decoding fails, we ignore it, and then we remove the quota annotation later. Is this desired?

Reply:
Thanks for catching this. Have updated (which was my original intention anyways) to not remove the annotation if decode could not get a valid quota.